File: Versions/v0-developing/src/OmniSphere.LiveStreamService/OmniSphere.LiveStreamService.Core/Interfaces/Repository/IKeyAccessRepository.cs
using OmniSphere.LiveStreamService.Core.Entity;

namespace OmniSphere.LiveStreamService.Core.Interfaces.Repository;

public interface IKeyAccessRepository
{
    Task AddAsync(KeyAccessEntity keyAccess);
    Task ExtendAsync(string keyAccess, int durationInMinutes);
    Task RemoveAsync(string keyAccess);
    Task<KeyAccessEntity> GetByKeyAccess(string keyAccess);
    Task<List<KeyAccessEntity>> GetAllByUserIdAsync(string userId);
}

File: Versions/v0-developing/src/KeyAccess/src/main/java/omnisphere/microsservices/KeyAccess/repository/IKeyAccessRepository.java
package omnisphere.microsservices.KeyAccess.repository;

import omnisphere.microsservices.KeyAccess.entity.KeyAccess;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

import java.util.List;

@Repository
public interface IKeyAccessRepository extends ReactiveMongoRepository<KeyAccess, String> {
    Mono<KeyAccess> findByKey(String key);
    Mono<List<KeyAccess>> findByUserId(String userId);
}
